projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/components/RunJob.tsx

```diff
@@ -71,41 +71,139 @@ export default class RunJobDialog extends Component<IJobPathProp> {
   };
 }
 
-export async function showRunJobDialog() {
+export async function showRunJobDialog(docManager?: IDocumentManager) {
   const result = await showDialog({
     title: 'Run Job',
-    body: (
-      <RunJobDialog
-        jobPath={jobData.get(VdkOption.PATH)!}
-      ></RunJobDialog>
-    ),
+    body: <RunJobDialog jobPath={jobData.get(VdkOption.PATH)!}></RunJobDialog>,
     buttons: [Dialog.okButton(), Dialog.cancelButton()]
   });
   if (result.button.accept) {
     let { message, status } = await jobRunRequest();
-        if (status) {
-          showDialog({
-            title: 'Run Job',
-            body: <div className='vdk-run-dialog-message-container'>
-            <p className='vdk-run-dialog-message'>The job was executed successfully!</p>
-          </div>,
-            buttons: [Dialog.okButton()]
-          });
-        }
-        else{
-          message = "ERROR : " + message;
-          const  errorMessage = new VdkErrorMessage(message);
-          showDialog({
-            title: 'Run Job',
-            body: <div  className="vdk-run-error-message ">
+    if (status) {
+      showDialog({
+        title: 'Run Job',
+        body: (
+          <div className="vdk-run-dialog-message-container">
+            <p className="vdk-run-dialog-message">
+              The job was executed successfully!
+            </p>
+          </div>
+        ),
+        buttons: [Dialog.okButton()]
+      });
+    } else {
+      message = 'ERROR : ' + message;
+      const errorMessage = new VdkErrorMessage(message);
+      if (
+        !docManager ||
+        !(await handleErrorsProducedByNotebookCell(errorMessage, docManager))
+      ) {
+        showDialog({
+          title: 'Run Job',
+          body: (
+            <div className="vdk-run-error-message ">
               <p>{errorMessage.exception_message}</p>
               <p>{errorMessage.what_happened}</p>
               <p>{errorMessage.why_it_happened}</p>
               <p>{errorMessage.consequences}</p>
               <p>{errorMessage.countermeasures}</p>
-            </div>,
-            buttons: [Dialog.okButton()]
-          });
-        }
+            </div>
+          ),
+          buttons: [Dialog.okButton()]
+        });
+      }
+    }
   }
 }
+
+export const findFailingCellId = (message: String): string => {
+  const regex = /cell_id:([0-9a-fA-F-]+)/;
+  const match = message.match(regex);
+  if (match) return match[1];
+  return '';
+};
+
+/**
+ * Seperate handling for notebook errors - option for the user to navigate to the failing cell when error is produced
+ */
+export const handleErrorsProducedByNotebookCell = async (
```

Comment:
Some general feedback on this function: It's a little hard to understand what's going on. There's a lot of room for improvement.

1. You can export the dialog templates to constants.
2. You can do away with at least some of the checks
3. You can avoid so much nesting by exporting some of the logic to separate functions.

Reply:
refactored